### DISCUSSION TITLE
MS SQL Server - Run Query Error

### DISCUSSION THREAD

**Discussion by Pgedela** [Budibase/budibase#2954] [Help]
![image](https://user-images.githubusercontent.com/92288941/136736974-f31af621-33fe-4aa3-a7c3-8dcb1f6a9e2e.png)

Hi All,

I am new to Budibase, self hosted today to trial the platform.
I am connecting to MS SQL Server as data source.

When I run the query, I get this error message (ref. screenshot)

Greatly appreciate your inputs !

**Comment by mjashanks** [comment]
Hey @Pgedela 

So, because you have not entered any default values for your parameters, your test query will look like

```sql
SELECT  FROM dbo.Client
```

If you enter the values

Field | Default Value
-------|-------------------
ClientID | 1234
OrganisationName | ACME Inc

Then your query will look like

```sql
SELECT 1234, ACME Inc  FROM dbo.Client
```

...which still doesn't seem right to me!

Maybe you mean your query to look more like

```sql
SELECT ClientID, OrganisationName FROM dbo.Client WHERE ClientId = {{ ClientId }}
```
which would result in 

```sql
SELECT ClientID, OrganisationName FROM dbo.Client WHERE ClientId = 1234
```

Our MS SQL connector is going to get an upgrade soon - so that it will work like internal tables. It will be able to read tables from your DB, and give yo all the basic INSERT, UPDATE, DELETE and SELECT by ID. This is how the Postgres and MySQL ones work at the moment. https://docs.budibase.com/guides/crud-app-with-sql

**Reply by Pgedela** [reply]
hey @mjashanks 
thanks for the reply ! 
And its great to hear that MS SQL connector is going to get an update soon !
I believe, it will be a big boost for enterprise customers who are still using Microsoft stack like us.

wrt the query,
its a simple query which clientid & the name from 'client' table (like this)
`SELECT ClientID, OrganisationName FROM dbo.Client`

what if I dont have any parameters, I am thinking this query should work without parameters.
But, it is still returning the same above error

**Comment by Pgedela** [comment]
@mjashanks I am wondering, if SQL Server is connected or not  ? which i might be causing this issue.
I cant tell if it is successfully connected, any idea how to ascertain this ?

**Reply by mjashanks** [reply]
Yea - that sounds like it might be it.

Note that if you are running Budibase **and** SQL Server locally, the SQL Server host will be `host.docker.internal` (i.e. instead of `localhost`).

If you are using Linux, it will be `172.17.0.1`
